ext/neural_snow/NeuralSnow.jl

```diff
@@ -0,0 +1,240 @@
+Helper function for evaluating the neural network in a pointwise manner over a `ClimaCore.Field`
+and returning the output in a broadcastable way.
+"""
+function eval_nn(
+    model,
+    z::FT,
+    swe::FT,
+    P::FT,
+    T::FT,
+    R::FT,
+    qrel::FT,
+    u::FT,
+)::FT where {FT}
+    input = FT.([z, swe, qrel, R, u, T, P])
+    return model(input)[1]
+end
+
+"""
+    dzdt(density::NeuralDepthModel, model::SnowModel{FT}, Y, p, t) where {FT}
+Returns the change in snow depth (rate) given the current model state and the `NeuralDepthModel`
+density paramterization, passing the approximate average of the forcings over the last 24 hours instead of
+the instantaneous value.
+"""
+function dzdt(density::NeuralDepthModel, Y)
+    return eval_nn.(
+        Ref(density.z_model),
+        Y.snow.Z,
+        Y.snow.S,
+        Y.snow.P_avg,
+        Y.snow.T_avg,
+        Y.snow.R_avg,
+        Y.snow.Qrel_avg,
+        Y.snow.u_avg,
+    )
+end
+
+"""
+    clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT
+A helper function which clips the tendency of Z such that
+its behavior is consistent with that of S.
+"""
+function clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT where {FT}
+    if (S + dSdt * Δt) <= eps(FT)
+        return -Z / Δt
+    elseif (Z + dZdt * Δt) < (S + dSdt * Δt)
+        #do we need to do something about setting q_l = 1 here?
+        return (S - Z) / Δt + dSdt
+    else
+        return dZdt
+    end
+end
+
+"""
+    update_density_prog!(density::NeuralDepthModel, model::SnowModel, Y, p)
+Updates all prognostic variables associated with density/depth given the current model state and the `NeuralDepthModel`
+density paramterization.
+"""
+function update_density_prog!(
+    density::NeuralDepthModel,
+    model::SnowModel,
+    dY,
+    Y,
+    p,
+)
+    #Get Inputs:
+    #is this too many allocations?
+    z = Y.snow.Z
+    swe = Y.snow.S
+    dswedt = p.snow.applied_water_flux #do i need to multiply by -1 here for directions?
+    dprecipdt_snow = abs.(p.drivers.P_snow) #need to change downward direction to scalar
+    air_temp = p.drivers.T .- model.parameters.earth_param_set.T_freeze
+    sol_rad = abs.(p.drivers.SW_d) #need to change downward direction to scalar
+    rel_hum =
+        Thermodynamics.relative_humidity.(
+            Ref(model.boundary_conditions.atmos.thermo_params),
+            p.drivers.thermal_state,
+        )
+    wind_speed = p.drivers.u
+    dt = model.parameters.Δt
+    β = density.α
+
+    dY.snow.Z .= clip_dZdt.(swe, z, dswedt, dzdt(density, Y), dt)
+    dY.snow.P_avg = β .* (dprecipdt_snow .- Y.snow.P_avg)
+    dY.snow.T_avg = β .* (air_temp .- Y.snow.T_avg)
+    dY.snow.R_avg = β .* (sol_rad .- Y.snow.R_avg)
+    dY.snow.Qrel_avg = β .* (rel_hum .- Y.snow.Qrel_avg)
+    dY.snow.u_avg = β .* (wind_speed .- Y.snow.u_avg)
+end
+
+end
+
+#do tests need to be made for these functions? since it is an extension?
```

Comment:
we have neural snow model extension tests currently:
https://github.com/CliMA/ClimaLand.jl/blob/main/test/standalone/Snow/tool_tests.jl

it would be good to add some tests for these functions